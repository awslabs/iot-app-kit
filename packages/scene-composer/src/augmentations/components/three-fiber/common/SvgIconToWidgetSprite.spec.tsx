import renderer from 'react-test-renderer';
import React from 'react';

import { DefaultAnchorStatus, SelectedAnchor } from '../../../..';
import {
  ErrorIconSvgString,
  InfoIconSvgString,
  SelectedIconSvgString,
  VideoIconSvgString,
  WarningIconSvgString,
} from '../../../../assets';

import svgIconToWidgetSprite from './SvgIconToWidgetSprite';

describe('svgIconToWidgetSprite', () => {
  const icons = [
    ['Selected', { key: SelectedAnchor, icon: SelectedIconSvgString }],
    ['Info', { key: DefaultAnchorStatus.Info, icon: InfoIconSvgString }],
    ['Warning', { key: DefaultAnchorStatus.Warning, icon: WarningIconSvgString }],
    ['Error', { key: DefaultAnchorStatus.Error, icon: ErrorIconSvgString }],
    ['Video', { key: DefaultAnchorStatus.Video, icon: VideoIconSvgString }],
  ];
  interface Icons {
    key: string;
    icon: string;
  }
  icons.forEach((value) => {
    it(`it should render the ${value[0]} correctly`, () => {
      jest.spyOn(window.Math, 'random').mockReturnValue(0.1);
      const { key, icon } = value[1] as Icons;
      const container = renderer.create(svgIconToWidgetSprite(icon, key, false, true));

      expect(container).toMatchSnapshot();
    });
  });

  icons.forEach((value) => {
    it(`it should render the always visible ${value[0]} correctly`, () => {
      jest.spyOn(window.Math, 'random').mockReturnValue(0.1);
      const { key, icon } = value[1] as Icons;
      const container = renderer.create(svgIconToWidgetSprite(icon, key, true, true));

      expect(container).toMatchSnapshot();
    });
  });

  icons.forEach((value) => {
    it(`it should render the constant sized ${value[0]} correctly`, () => {
      jest.spyOn(window.Math, 'random').mockReturnValue(0.1);
      const { key, icon } = value[1] as Icons;
      const container = renderer.create(svgIconToWidgetSprite(icon, key, false, false));

      expect(container).toMatchSnapshot();
    });
  });
});
