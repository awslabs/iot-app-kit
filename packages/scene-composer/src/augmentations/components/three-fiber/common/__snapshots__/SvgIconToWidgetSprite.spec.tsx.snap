// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgIconToWidgetSprite it should render the Error correctly 1`] = `
<widgetSprite
  name="Error"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the Info correctly 1`] = `
<widgetSprite
  name="Info"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the Selected correctly 1`] = `
<widgetSprite
  name="Selected"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the Video correctly 1`] = `
<widgetSprite
  name="Video"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the Warning correctly 1`] = `
<widgetSprite
  name="Warning"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Error correctly 1`] = `
<widgetSprite
  name="Error"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Info correctly 1`] = `
<widgetSprite
  name="Info"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Selected correctly 1`] = `
<widgetSprite
  name="Selected"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Video correctly 1`] = `
<widgetSprite
  name="Video"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Warning correctly 1`] = `
<widgetSprite
  name="Warning"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
    <sprite
      renderOrder={10}
    >
      <spriteMaterial
        depthFunc={6}
        opacity={0.5}
      />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the constant sized Error correctly 1`] = `
<widgetSprite
  name="Error"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the constant sized Info correctly 1`] = `
<widgetSprite
  name="Info"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the constant sized Selected correctly 1`] = `
<widgetSprite
  name="Selected"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the constant sized Video correctly 1`] = `
<widgetSprite
  name="Video"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
  </group>
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the constant sized Warning correctly 1`] = `
<widgetSprite
  name="Warning"
>
  <group
    attach="visual"
  >
    <sprite
      renderOrder={10}
    >
      <spriteMaterial />
    </sprite>
  </group>
</widgetSprite>
`;
